QEP-auxiliary-storage.rst

```diff
@@ -0,0 +1,178 @@
+between the two tables is not possible, leading to a solution not satisfactory.
+
+We propose here a more integrated solution where the creation of auxiliary columns or layers
+are brought more transparently to the user by storing them in a central repository carried by the
+project file.
+
+Requirements
+------------
+
+The proposed solution have the following requirements :
+
+- It should allow to transparently add columns of any type to a layer, without adding them to the source data source.
+Beside usual types (string, real, integer), we may also think of geometry types. We only consider types that can be converted to atomic types (string for WKT or WKB)
+The user should be able to edit values in these columns when the layer is in editing mode.
+
+- These values should not be stored with the source data, and should be carried by the project, i.e.
+when the project file is moved or copied, the auxiliary data should still be there.
+
+- Adding a new row to the layer also adds a new row for each auxiliary field (with a null or default value)
+
+- Removing a row from the layer also removes the corresponding row of each auxiliary field
+
+- New GUI elements should ease the creation of auxiliary fields
+  - on each "data-defined" property
+  - possibly on groups of "data-defined": for manual label placement or label connectors
+
+- It should be possible to save auxiliary data to a new layer
+
+Proposed solution
+-----------------
+
+We propose to add a new type of field, close to the existing "virtual field" (where a field can be defined as an expression).
+Here a new type of field, "auxiliary field", will point to a central data storage.
+
+The central data storage should allow to store any kind of auxiliary column definition and data for any layer. And it should
+be linked to, or be part of, a project file. The access to auxiliary data should be efficient.
+
+Core changes
+------------
+
+Project file
+------------
+
+The current project file format (XML) is too limited to carry an efficient storage of such auxiliary data. It should somehow be evolved
+to include a binary indexable format. SQLite is chosen as a first candidate for this data format. But it might evolve if performance issues
+are found.
+
+Given the current state of the core code where lots of classes assume the project file is an XML file (everything that is saved to the project file
+has a writeXML method), it seems hard to completely refactor the project file format to a format based on SQLite.
+
+We propose to use a Zip file as the root of the new project file format. A Zip file exposes a filesystem-like structure.
+The original XML-based project file will then be embedded in the Zip file. And additional files could de added, and in particular
+an SQLite file.
+It only adds a step after actually saving the XML tree: including it in a Zip file. Symetrically during the loading, the first step would be to extract the XML file and pass it to the
+existing readXML methods.
+
```

Comment:
What's the expected timeframe for this? Changing the project file format sounds like a 3.0 feature to me.